text/entities/0256-entities-data-model.md

```diff
@@ -0,0 +1,790 @@
+receivers to infer that a no longer reported entity is gone, even if the
+EntityDelete event was not observed. Optional, if missing the interval
+is unknown.
+    </td>
+   </tr>
+   <tr>
+    <td>Type
+    </td>
+    <td>
+    </td>
+    <td>See data model.
+
+</td>
+   </tr>
+   <tr>
+    <td>Id
+    </td>
+    <td>
+    </td>
+    <td>See data model
+</td>
+   </tr>
+   <tr>
+    <td>Attributes
+    </td>
+    <td>
+    </td>
+    <td>See data model
+</td>
+   </tr>
+</table>
+
+We say that an entity mutates (changes) when one or more of its
+descriptive attributes changes. A new descriptive attribute may be
+added, an existing descriptive attribute may be deleted or a value of an
+existing descriptive attribute may be changed. All these changes
+represent valid mutations of an entity over time. When these mutations
+happen the identity of the entity does not change.
+
+When the entity's state is changed it is expected that the source will
+emit a new EntityState event with a fresh timestamp and full list of
+values of all other fields.
+
+Entity event producers are recommended to periodically emit events even
+if the entity does not change. In this case the Type, Id and Attribute
+fields will remain the same, but a fresh Timestamp will be recorded in
+the event. Producing such events allows the system to be resilient to
+event losses. Even if some events are lost eventually the correct state
+of the entity is more likely to be delivered to the final destination.
+Periodic sending of EntityState events also serves as a liveliness
+indicator (see below how it can be used in lieu of EntityDelete event).
+
+### EntityDelete Event
+
+EntityDelete event indicates that a particular entity is gone:
+
+<table>
+   <tr>
+    <td><strong>Field</strong>
+    </td>
+    <td><strong>Type</strong>
+    </td>
+    <td><strong>Description</strong>
+    </td>
+   </tr>
+   <tr>
+    <td>Timestamp
+    </td>
+    <td>nanoseconds
+    </td>
+    <td>The time when the entity was deleted. The time is measured by
+the origin clock. The field is required.
+    </td>
+   </tr>
+   <tr>
+    <td>Type
+    </td>
+    <td>
+    </td>
+    <td>See data model
+</td>
+   </tr>
+   <tr>
+    <td>Id
+    </td>
+    <td>
+    </td>
+    <td>See data model
+</td>
+   </tr>
+</table>
+
+Note that EntityDelete is optional and is not guaranteed to be sent when
+the entity is gone. Recipients of entity signals should be prepared to
+handle this situation by expiring entities that are no longer seeing
+EntityState events reported (i.e. treat the presence of EntityState
+events as a liveliness indicator).
+
+The expiration mechanism is based on the previously reported `Interval` field of
+EntityState event. The recipient can use this value to compute when to expect the next
+EntityState event and if the event does not arrive in a timely manner (plus some slack)
+it can consider the entity to be gone even if the EntityDelete event was not observed.
```

Comment:
What should be done to EntityState values received and post-dated to an EntityDelete value?

Reply:
That indicates an entity that was gone and came back. It is a valid situation.